src/main/java/io/aiven/kafka/auth/json/AivenAcl.java

```diff
@@ -154,10 +176,22 @@ private boolean comparePattern(final Pattern p1, final Pattern p2) {
         return p1.toString().equals(p2.toString());
     }
 
+    public void validate() {
+        // check that only one of resources fields is not null
+        // in the future we can add checks for other fields
+        final long resourceCount = Stream.of(resourceRe, resourceRePattern, resourceLiteral, resourcePrefix)
+                .filter(Objects::nonNull)
+                .count();
+        if (resourceCount != 1) {
+            throw new IllegalArgumentException("Must specify exactly one \"resource\" for " + this);
```

Comment:
This is problematic because it is not catched. JSON failures are handled by `loadAcls` simply using empty list as effective ACLs, but with this exception something unknown will happen.